crates/turborepo-filewatch/src/lib.rs

```diff
@@ -454,35 +514,80 @@ async fn wait_for_cookie(
 ) -> Result<(), WatchError> {
     // TODO: should this be passed in? Currently the caller guarantees that the
     // directory is empty, but it could be the responsibility of the
-    // filewatcher...
+    // filewatcher..
     let cookie_path = cookie_dir.join_component(".turbo-cookie");
-    cookie_path.create_with_contents("cookie").map_err(|e| {
-        WatchError::Setup(format!("failed to write cookie to {}: {}", cookie_path, e))
-    })?;
-    loop {
-        let event = tokio::time::timeout(Duration::from_millis(2000), recv.recv())
-            .await
-            .map_err(|e| WatchError::Setup(format!("waiting for cookie timed out: {}", e)))?
-            .ok_or_else(|| {
-                WatchError::Setup(
-                    "filewatching closed before cookie file  was observed".to_string(),
-                )
-            })?
-            .map_err(|err| {
-                WatchError::Setup(format!("initial watch encountered errors: {}", err))
-            })?;
-        if event.paths.iter().any(|path| {
-            let path: &Path = path;
-            path == (&cookie_path as &AbsoluteSystemPath)
-        }) {
-            // We don't need to stop everything if we failed to remove the cookie file
-            // for some reason. We can warn about it though.
-            if let Err(e) = cookie_path.remove() {
-                warn!("failed to remove cookie file {}", e);
-            }
-            return Ok(());
+
+    tracing::debug!("Creating cookie file at: {}", cookie_path);
+
+    // Check device information for debugging cross-volume issues
+    #[cfg(target_os = "macos")]
+    {
+        if let Ok(cookie_dir_metadata) = std::fs::metadata(cookie_dir.as_std_path()) {
+            use std::os::unix::fs::MetadataExt;
+            let device_id = cookie_dir_metadata.dev();
+            tracing::debug!("Cookie directory device ID: {}", device_id);
         }
     }
+
+    std::fs::write(&cookie_path, "turbo")?;
+
+    tracing::debug!("Cookie file created, waiting for filesystem event");
+
+    // Dynamic timeout based on filesystem characteristics
+    let timeout_duration = if cookie_dir.to_string_lossy().starts_with("/Volumes/") {
+        // External volume detected - use longer timeout
+        tracing::debug!("External volume detected, using extended cookie timeout (6 seconds)");
+        Duration::from_secs(6)
```

Comment:
While I was testing, it was possible for an external volume did seem to be slightly slower. 6 is an arbitrary number; we can lower it if we want. I moved it up until I stopped having problems.